apps/main/src/loan/entities/mint-markets.ts

```diff
@@ -0,0 +1,39 @@
+import { useMemo } from 'react'
+import { isAddress } from 'viem'
+import { ChainId } from '@/loan/types/loan.types'
+import { fromEntries } from '@curvefi/prices-api/objects.util'
+import { useConnection } from '@ui-kit/features/connect-wallet'
+import type { Address } from '@ui-kit/utils'
+
+/**
+ * Hook to get a mapping of mint market controller address to market name for all mint markets on a specific chain.
+ * Primarily useful fetching mint markets via URL.
+ */
+export const useMintMarketMapping = ({ chainId }: { chainId: ChainId | undefined }) => {
```

Comment:
Note how all these things are not async, hence not bothering with Tanstack. I'm afraid it won't be this easy for `lend` markets.